### PR TITLE
Add third-party tests for BeanieODM

### PR DESCRIPTION
**Pull request by mdaffad**
## Change Summary

Integrate beanie tests on third party test workflow, part of third party tests on #11160.
Workflow result - https://github.com/mdaffad/pydantic/actions/runs/12477146497

## Related issue number

#11160

## Checklist

* [ ] The pull request title is a good summary of the changes - it will be used in the changelog
* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [ ] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: .github/workflows/third-party.yml
Comment (Viicos) [inline .github/workflows/third-party.yml:149]:
```suggestion
        mongodb-version: '8.0'
```

Let's test against a single MongoDB version.

File: .github/workflows/third-party.yml
Comment (Viicos) [inline .github/workflows/third-party.yml:129]:
```suggestion
```

File: .github/workflows/third-party.yml
Comment (Viicos) [inline .github/workflows/third-party.yml:144]:
```suggestion
    - uses: actions/setup-python@v5
      with:
        python-version: ${{ matrix.python-version }}
```

Best to be as close to the project's CI.

File: .github/workflows/third-party.yml
Comment (Viicos) [inline .github/workflows/third-party.yml:131]:
```suggestion
    - name: Checkout Beanie
```

File: .github/workflows/third-party.yml
Comment (Viicos) [inline .github/workflows/third-party.yml:156]:
```suggestion
    - name: Install Beanie dependencies
      run: |
        pip install .[test,ci]
        pip uninstall --yes pydantic 
        pip install ./pydantic-latest
```

File: .github/workflows/third-party.yml
Comment (Viicos) [inline .github/workflows/third-party.yml:163]:
```suggestion
    - name: Run Beanie tests
```

**Review by Viicos — CHANGES_REQUESTED** [review]
Thanks for the PR. This looks great

File: .github/workflows/third-party.yml
Comment (sydney-runkle) [inline .github/workflows/third-party.yml:139]:
```suggestion

```

**Comment by sydney-runkle** [comment]
LGTM, let's wait for @Viicos' approval before we merge :)

**Review by Viicos — APPROVED** [review]
Thanks!

**Pull request merged**
